admin/lib/validator/pupil-validator.js

```diff
@@ -133,7 +133,8 @@ module.exports.validate = async (pupilData, schoolId, isMultiplePupilsSubmission
   if (!isEmpty(upn) && expected !== upn[0]) {
     upnErrorArr.push(addPupilErrorMessages.upnInvalidCheckDigit)
   }
-  if (!isEmpty(upn) && !upnService.hasValidLaCode(upn)) {
+  const isValidLaCode = await upnService.hasValidLaCode(upn)
+  if (!isEmpty(upn) && !isValidLaCode) {
```

Comment:
wouldn't it be less expensive to test for empty first?

Reply:
I don't think so.  All the tests are applied before returning the results to the user.

Reply:
yep, just opened the file and can see that now. thanks